bool front
bool rear
bool middle